src/main/java/uk/gov/hmcts/cmc/claimstore/events/claim/PostClaimOrchestrationHandler.java

```diff
@@ -36,33 +66,75 @@ public PostClaimOrchestrationHandler(
+                : claim;
+
+        uploadClaimIssueReceiptOperation = (claim, event, docs) ->
+            isUploadClaimReceiptSuccess.test(claim.getClaimSubmissionOperationIndicators())
+                ? uploadOperationService.uploadDocument(claim, event.getAuthorisation(), docs.getClaimIssueReceipt(),
+                CaseEvent.CLAIM_ISSUE_RECEIPT_UPLOAD)
+                : claim;
+
+        rpaOperation = (claim, event, docs) ->
+            isRpaOperationSuccess.test(claim.getClaimSubmissionOperationIndicators())
+                ? rpaOperationService.notify(claim, event.getAuthorisation(), docs.getSealedClaim())
+                : claim;
+
+        notifyStaffOperation = (claim, event, docs) ->
+            isNotifyStaffSuccess.test(claim.getClaimSubmissionOperationIndicators())
+                ? notifyStaffOperationService.notify(claim, event.getAuthorisation(), docs.getSealedClaim(),
+                docs.getDefendantLetter())
+                : claim;
+
+        notifyCitizenOperation = (claim, event, docs) ->
+            isNotifyCitizenSuccess.test(claim.getClaimSubmissionOperationIndicators())
+                ? claimantOperationService.notifyCitizen(claim, event.getSubmitterName(), event.getAuthorisation())
+                : claim;
+
+        notifyRepresentativeOperation = (claim, event, docs) ->
+            isNotifyCitizenSuccess.test(claim.getClaimSubmissionOperationIndicators())
+                ? claimantOperationService.confirmRepresentative(claim, event.getSubmitterName(),
+                ((RepresentedClaimCreatedEvent) event).getRepresentativeEmail(), event.getAuthorisation())
+                : claim;
     }
 
     @EventListener
     public void citizenIssueHandler(CitizenClaimCreatedEvent event) {
         try {
             Claim claim = event.getClaim();
             String authorisation = event.getAuthorisation();
-            String submitterName = event.getSubmitterName();
-
-            Claim updatedClaim = pinOrchestrationService.process(claim, authorisation, submitterName);
+            Claim updatedClaim;
 
-            PDF sealedClaimPdf = documentOrchestrationService.getSealedClaimPdf(claim);
-            updatedClaim = uploadOperationService.uploadDocument(updatedClaim, authorisation, sealedClaimPdf);
+            GeneratedDocuments generatedDocuments =
+                documentOrchestrationService.generateForCitizen(claim, authorisation);
 
-            PDF claimIssueReceiptPdf = documentOrchestrationService.getClaimIssueReceiptPdf(claim);
-            updatedClaim = uploadOperationService.uploadDocument(updatedClaim, authorisation, claimIssueReceiptPdf);
+            updatedClaim = CompletableFuture
+                .supplyAsync(() -> performPinOperations.perform(claim, event, generatedDocuments))
+                .thenApplyAsync(pinCompletedClaim
+                    -> uploadSealedClaimOperation.perform(pinCompletedClaim, event, generatedDocuments))
+                .thenApplyAsync(uploadedSealedClaim
```

Comment:
Minor Suggestion: maybe change this to thenApply.

Reply:
Good point. But CompletableFuture is misunderstood here. So falling back to Function.